src/test/scala/clump/ClumpSourceSpec.scala

```diff
@@ -20,8 +20,40 @@ class ClumpSourceSpec extends Spec {
     val repo = smartMock[TestRepository]
   }
 
+  "allows to create a clump source (ClumpSource.from)" >> {
+    "set input" in {
+      def fetch(inputs: Set[Int]) = Future.value(inputs.map(i => i -> i.toString).toMap)
+      val source = ClumpSource.from(fetch)
+      clumpResult(source.get(1)) mustEqual Some("1")
+    }
+    "list input" in {
+      def fetch(inputs: List[Int]) = Future.value(inputs.map(i => i -> i.toString).toMap)
+      val source = ClumpSource.from(fetch)
+      clumpResult(source.get(1)) mustEqual Some("1")
+    }
+  }
+
+  "allows to create a clump source with key function (ClumpSource.apply)" >> {
+    "set input" in {
+      def fetch(inputs: Set[Int]) = Future.value(inputs.map(_.toString))
+      val source = ClumpSource.apply(fetch)(_.toInt)
```

Comment:
Stylistic again, but should we just use `ClumpSource(fetch)` instead us explicitly `.apply`